pandas/errors/__init__.py

```diff
@@ -91,6 +91,96 @@ class PerformanceWarning(Warning):
     """
 
 
+class PandasChangeWarning(Warning):
+    """
+    Warning raised for any an upcoming change.
+
+    See Also
+    --------
+    errors.Pandas4Warning : Class for deprecations to be enforced in pandas 4.0.
+
+    Examples
+    --------
+    >>> pd.errors.PandasChangeWarning
+    <class 'pandas.errors.PandasChangeWarning'>
+    """
+
+
+class PandasPendingDeprecationWarning(PandasChangeWarning, PendingDeprecationWarning):
+    """
+    Warning raised for an upcoming change that is a PendingDeprecationWarning.
+
+    See Also
+    --------
+    errors.Pandas4Warning : Class for deprecations to be enforced in pandas 4.0.
+
+    Examples
+    --------
+    >>> pd.errors.PandasPendingDeprecationWarning
+    <class 'pandas.errors.PandasPendingDeprecationWarning'>
+    """
+
+
+class PandasDeprecationWarning(PandasChangeWarning, DeprecationWarning):
+    """
+    Warning raised for an upcoming change that is a DeprecationWarning.
+
+    See Also
+    --------
+    errors.Pandas4Warning : Class for deprecations to be enforced in pandas 4.0.
+
+    Examples
+    --------
+    >>> pd.errors.PandasDeprecationWarning
+    <class 'pandas.errors.PandasDeprecationWarning'>
+    """
+
+
+class PandasFutureWarning(PandasChangeWarning, FutureWarning):
+    """
+    Warning raised for an upcoming change that is a FutureWarning.
+
+    See Also
+    --------
+    errors.Pandas4Warning : Class for deprecations to be enforced in pandas 4.0.
+
+    Examples
+    --------
+    >>> pd.errors.PandasFutureWarning
+    <class 'pandas.errors.PandasFutureWarning'>
+    """
+
+
+class Pandas4Warning(PandasDeprecationWarning):
+    """
+    Warning raised for an upcoming change that will be enforced in pandas 4.0.
+
+    See Also
+    --------
+    errors.Pandas4Warning : Class for deprecations to be enforced in pandas 4.0.
+
+    Examples
+    --------
+    >>> pd.errors.Pandas4Warning
+    <class 'pandas.errors.Pandas4Warning'>
+    """
+
+
+class Pandas5Warning(PandasPendingDeprecationWarning):
+    """
+    Warning raised for an upcoming change that will be enforced in pandas 5.0.
+
+    See Also
+    --------
+    errors.Pandas4Warning : Class for deprecations to be enforced in pandas 4.0.
+
+    Examples
+    --------
+    >>> pd.errors.Pandas5Warning
+    <class 'pandas.errors.Pandas5Warning'>
```

Comment:
I wonder if we should have an abstract base class `PandasWarning` that has an abstract class method `version()`, and then both `Pandas4Warning` and `Pandas5Warning` inherit from that class as well, and define `version()` to return 4 and 5 respectively. 

Then in `deprecate_nonkeyword_arguments()`, instead of checking for the class, you just check for `version()`.  See comment there.